compiler/rustc_infer/src/infer/opaque_types.rs

```diff
@@ -146,7 +146,11 @@ impl<'tcx> InferCtxt<'tcx> {
                         }
                     }
                     DefiningAnchor::Bubble => {}
-                    DefiningAnchor::Error => return None,
+                    DefiningAnchor::Error => {
+                        if !self.intercrate {
```

Comment:
Aren't we sometimes emitting alias-relate goals in coherence? Or we planned to do so? Perhaps we could just do that and assert against `self.intercrate` in this function.

Reply:
We're def not doing that right now. I presumed we didn't want to touch old coherence anymore and just wait until it gets replaced by the new solver, which doesn't care about the bubble/error distinction anyway.

Reply:
can you change it to match on `DefiningAnchor` only if `intercrate` is false instead? Only having the intercrate check in the `Error` branch feels weird :thinking: